vowpalwabbit/example.cc

```diff
@@ -12,7 +12,13 @@ VW_WARNING_DISABLE_DEPRECATED_USAGE
 example::example()
 {
   memset(&l, 0, sizeof(polylabel));
-  memset(&pred, 0, sizeof(polyprediction));
+  // init predictions
```

Comment:
Would it be possible to put these in the polypred constructor/destructor and just call placement new/the destructor here? Or even better, initialize everything properly in the constructor instead of relying on memset

Reply:
Let me fix the memory leaks find and get this PR green, then I'll address this issue.

Let's keep cleanups and improvements in the context of this PR minimal to simplify any bisection we need to do it the future.